tests/validate/bandwidth_test/src/host.cpp

```diff
@@ -151,102 +161,175 @@ int main(int argc, char** argv) {
 
     double max_throughput = 0;
     int reps = stoi(iter_cnt);
+    if (num_kernel_ddr) {
+        //Starting at 4K and going up to 16M with increments of power of 2
+        for (uint32_t i = 4 * 1024; i <= 16 * 1024 * 1024; i *= 2) {
+            unsigned int data_size = i;
+
+            if (xcl::is_emulation()) {
+                reps = 2; // reducing the repeat count to 2 for emulation flow
+                // Running only upto 8K for emulation flow
+                if (data_size > 8 * 1024) {
+                    break;
+                }
+            }
 
-    for (uint32_t i = 4 * 1024; i <= 16 * 1024 * 1024; i *= 2) {
-        unsigned int DATA_SIZE = i;
+            unsigned int vector_size_bytes = data_size;
+            std::vector<unsigned char, aligned_allocator<unsigned char> > input_host(data_size);
+            std::vector<unsigned char, aligned_allocator<unsigned char> > output_host[num_kernel_ddr];
 
-        if (xcl::is_emulation()) {
-            reps = 2;
-            if (DATA_SIZE > 8 * 1024) break;
-        }
+            for (int i = 0; i < num_kernel_ddr; i++) {
+                output_host[i].resize(data_size);
+            }
+            for (uint32_t j = 0; j < data_size; j++) {
```

Comment:
Comment here would be nice here as well.  Something like:
`// Filling up memory with an incremental byte pattern`